File: November/Day6/899._Orderly_Queue.java
class Solution {
    public String orderlyQueue(String s, int k) {
        if(k>1){
            char tempArray[] = s.toCharArray();
            Arrays.sort(tempArray);
            return new String(tempArray);
        }
            
        int n=s.length();
        String doublestring=s+s;
        String ans=s;
        for(int i=1;i<n;i++){
            String sub=doublestring.substring(i,n+i);
            if(ans.compareTo(sub)>0)ans=sub;
            
        }
        return ans;
        
    }
}

File: November/Day6/899._Orderly_Queue.cpp
class Solution {
public:
    string orderlyQueue(string s, int k) {
        
        if (k==1)
        {
            int n=s.size();
            string doublestring=s+s;
            string ans=s;
            for (int i=1;i<n;i++) 
            {
                string str=doublestring.substr(i,n);
                if (str<ans) 
                {
                    ans = str;
                }
            }
            return ans;
            
        }
        sort(s.begin(),s.end());
        return s;
        
    }
};